### PR TITLE
DHFPROD-8504: Widget— Membership

### PR DESCRIPTION
**Pull request by iveyMarklogic**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [ ] Reviewed Tests
- [ ] Added to Release Wiki

### CODE REVIEW

File: examples/entity-viewer-v2/entity-viewer-client/src/main/ml-modules/root/explore-data/ui-config/config.json [hunk 1]
```diff
@@ -633,8 +633,29 @@
...
     },
 
-    "occupations": {}
+    "occupations": {},
 
+    "membership": {
+      "component": "Relationships",
+      "config": {
+        "arrayPath": "person.memberships.membership",
+        "iconSize": 35,
```
Comment (wooldridge) [inline examples/entity-viewer-v2/entity-viewer-client/src/main/ml-modules/root/explore-data/ui-config/config.json:642]: Let's add a date configuration here:
```
"dateFormat": "yyyy-MM-dd",
```
Then we can use it with our DateTime component.

File: marklogic-data-hub-central/ui-custom/src/components/Membership/Membership.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    let memberships = getValByConfig(data, config);
+    const items = config?.lists?.map((list, index) => {
+      const membership = memberships?.find(el => el?.list.toLowerCase() === list.toLowerCase());
+      const important = membership?.important === "true";
+      const classes = important ? "item highlighted" : membership ? "item success" : "item";
+      return (
+        <div key={index} className={classes}>
+          <span className="title">{list}</span>
+          <span className="icon" data-testid="icon-container">
+            {membership ? <Check data-testid="success-icon" color="#97e7d7" size={iconSize} /> : <X data-testid="error-icon" color="#a2a2a2" size={iconSize} />}
```
Comment (wooldridge) [inline marklogic-data-hub-central/ui-custom/src/components/Membership/Membership.tsx:54]: If you import our DateTime component, you can use it here for formatting:
```
{membership ? <span className="date"><DateTime config={{"format": props.config.dateFormat}}>{membership.ts}</DateTime></span> : <span className="empty date"></span>}
```

File: marklogic-data-hub-central/ui-custom/src/components/Membership/Membership.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+          "list10",
+          "list11",
+          "list12"
+        ]
+      }
+    }
+ */
+const Membership: React.FC<Props> = (props) => {
+  const {data, config} = props;
+  const iconSize = config.iconSize ? config.iconSize : 30
```
Comment (wooldridge) [inline marklogic-data-hub-central/ui-custom/src/components/Membership/Membership.tsx:43]: Allow configuration of item container style:
```
let membershipStyle: any = props.style ? props.style : props.config?.style ? props.config.style : {};
```
Then apply the style in the item DIV:
```
<div key={index} className={classes} style={membershipStyle}>
```

File: marklogic-data-hub-central/ui-custom/src/components/Membership/Membership.scss [hunk 4]
```diff
@@ -0,0 +1,60 @@
+$background-color: #f4f4f4;
+$border-color: #ebebeb;
+$highlighted-color: #97e7d7;
+$title-color: #a2a2a2;
+
+.Membership {
+    display: flex;
+    flex-wrap: wrap;
+    .item {
+        flex-basis: 140px;
```
Comment (wooldridge) [inline marklogic-data-hub-central/ui-custom/src/components/Membership/Membership.scss:10]: Let's make the default of 12 lists fit. Let's also constrain the item container size so it doesn't get messy:
```
        flex-basis: 8.3%;
        min-width: 110px; 
        max-width: 160px; 
```

File: examples/entity-viewer-v2/entity-viewer-client/src/main/ml-modules/root/explore-data/ui-config/config.json (hunk 1 above)
Comment (wooldridge) [inline examples/entity-viewer-v2/entity-viewer-client/src/main/ml-modules/root/explore-data/ui-config/config.json:642]: Let's tighten the height:
```
"iconSize": 32,
```

File: marklogic-data-hub-central/ui-custom/src/components/Membership/Membership.scss [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        display: flex;
+        flex-direction: column;
+        align-items: center;
+        border: 1px solid $border-color;
+        & > * {
+            padding: 10px 0px;
```
Comment (wooldridge) [inline marklogic-data-hub-central/ui-custom/src/components/Membership/Membership.scss:16]: Let's tighten the height:
```
padding: 8px 0px;
```

File: marklogic-data-hub-central/ui-custom/src/views/Detail.tsx [hunk 6]
```diff
@@ -94,54 +95,62 @@ const Detail: React.FC<Props> = (props) => {
...
 
-          <div className="row">
-            <div className="col-lg-7">
-
-              {config?.detail?.personal ? 
-                <Section title="Personal Info">
-                  {getPersonalItems(config?.detail?.personal?.items)}
+            {config?.detail?.membership?.config && <div className="row">
+              <div className="col-12">
+                <Section title="Membership" config={{
```
Comment (wooldridge) [inline marklogic-data-hub-central/ui-custom/src/views/Detail.tsx:108]: Let's tighten the height:
```
                <Section title="Membership" config={{
                  "headerStyle": {
                    "backgroundColor": "transparent"
                  },
                  "mainStyle": {
                    "padding-top": "8px"
                  }
                }}>
```

**Review by wooldridge — CHANGES_REQUESTED** [review]
Great job! I'm suggesting changes to make it more configurable and to tighten the height.

**Comment by iveyMarklogic** [comment]
@wooldridge I did all of the changes, Can you check it again?

**Review by wooldridge — APPROVED** [review]
Looks great, thanks!

**Pull request merged**